src/connector/src/parser/additional_columns.rs

```diff
@@ -41,73 +52,148 @@ fn kafka_compatible_column_vec() -> Vec<(&'static str, CompatibleAdditionalColum
     vec![
         (
             "key",
-            Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
-                ColumnCatalog {
-                    column_desc: ColumnDesc::named_with_additional_column(
-                        name,
-                        id,
-                        DataType::Bytea,
-                        AdditionalColumnType::Key,
-                    ),
-                    is_hidden: false,
-                }
-            }),
+            Box::new(
+                |id: ColumnId,
+                 name: &str,
+                 _inner_field: Option<&str>,
+                 _data_type: Option<PbDataType>|
```

Comment:
Hmmm, it become more hard to read as more arguments are added. See https://github.com/risingwavelabs/risingwave/pull/13707#discussion_r1431284839

Reply:
I have refactored this part and unified catalog generation.

Reply:
Thanks! Looks much better!